index.js

```diff
@@ -534,29 +534,34 @@ app.post(
 app.use(braveAlerter.getRouter())
 
 // Handler for incoming XeThru POST requests
-app.post('/api/xethru', Validator.body(['locationid', 'state', 'rpm', 'mov_f', 'mov_s']).exists(), async (req, res) => {
+app.post('/api/xethru', Validator.body(['coreid', 'state', 'rpm', 'mov_f', 'mov_s']).exists(), async (req, res) => {
   try {
     const validationErrors = Validator.validationResult(req).formatWith(helpers.formatExpressValidationErrors)
 
     if (validationErrors.isEmpty()) {
-      const { locationid, state, rpm, distance, mov_f, mov_s } = req.body
-
-      await redis.addXeThruSensorData(locationid, state, rpm, distance, mov_f, mov_s)
+      const { coreid, state, rpm, distance, mov_f, mov_s } = req.body
+      const location = await db.getLocationFromParticleCoreID(coreid)
+      if (!location) {
+        const errorMessage = `Bad request to ${req.path}: no location matches the coreID ${coreid}`
+        helpers.logError(errorMessage)
+        // Must send 200 so as not to be throttled by Particle (ref: https://docs.particle.io/reference/device-cloud/webhooks/#limits)
+        res.status(200).json(errorMessage)
+      } else {
+        await redis.addXeThruSensorData(location.locationid, state, rpm, distance, mov_f, mov_s)
 
-      const location = await db.getLocationData(locationid)
-      if (location.isActive && !location.heartbeatSentAlerts) {
-        await StateMachine.getNextState(location, handleAlert)
+        if (location.isActive && !location.heartbeatSentAlerts) {
+          await StateMachine.getNextState(location, handleAlert)
+        }
+        res.status(200).json('OK')
       }
-
-      res.status(200).json('OK')
     } else {
       const errorMessage = `Bad request to ${req.path}: ${validationErrors.array()}`
       helpers.logError(errorMessage)
       // Must send 200 so as not to be throttled by Particle (ref: https://docs.particle.io/reference/device-cloud/webhooks/#limits)
       res.status(200).json(errorMessage)
     }
   } catch (err) {
-    const errorMessage = `Error calling ${req.path}: ${JSON.stringify(err)}`
+    const errorMessage = `Error calling ${req.path}: ${err.toString()}`
```

Comment:
woops! This is why we get those Error calling /api/xethru {} messages isn't it.

Reply:
Yup, exactly.